app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt

```diff
@@ -0,0 +1,399 @@
+              )
+            )
+          )
+        )
+
+      onView(withId(R.id.profile_type_supervisor_navigation_card))
+        .check(
+          matches(
+            allOf(
+              isDisplayed(),
+              hasDescendant(
+                withText(R.string.onboarding_profile_type_activity_parent_text)
+              )
+            )
+          )
+        )
+    }
+  }
+
+  @Test
+  fun testFragment_portrait_stepCountTextIsDisplayed() {
+    launchOnboardingProfileTypeActivity().use {
+      onView(withId(R.id.onboarding_steps_count))
+        .check(
+          matches(
+            allOf(
+              isDisplayed(),
+              withText(
+                R.string.onboarding_step_count_two
+              )
+            )
+          )
+        )
+    }
+  }
+
+  @Test
+  fun testFragment_studentNavigationCardClicked_launchesCreateProfileScreen() {
+    launchOnboardingProfileTypeActivity().use {
+      onView(withId(R.id.profile_type_learner_navigation_card)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+      // Does nothing for now, but should fail once navigation is implemented in a future PR.
+      onView(withId(R.id.profile_type_learner_navigation_card))
+        .check(matches(isDisplayed()))
+
+      onView(withId(R.id.profile_type_supervisor_navigation_card))
+        .check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testFragment_orientationChange_studentNavigationCardClicked_launchesCreateProfileScreen() {
+    launchOnboardingProfileTypeActivity().use {
+      onView(isRoot()).perform(orientationLandscape())
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.profile_type_learner_navigation_card)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+      // Does nothing for now, but should fail once navigation is implemented in a future PR.
+      onView(withId(R.id.profile_type_learner_navigation_card))
+        .check(matches(isDisplayed()))
+
+      onView(withId(R.id.profile_type_supervisor_navigation_card))
+        .check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testFragment_supervisorNavigationCardClicked_launchesProfileChooserScreen() {
+    launchOnboardingProfileTypeActivity().use {
+      onView(withId(R.id.profile_type_supervisor_navigation_card)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+      intended(hasComponent(ProfileChooserActivity::class.java.name))
+    }
+  }
+
+  @Test
+  fun testFragment_orientationChange_supervisorCardClicked_launchesProfileChooserScreen() {
+    launchOnboardingProfileTypeActivity().use {
+      onView(isRoot()).perform(orientationLandscape())
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.profile_type_supervisor_navigation_card)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+      intended(hasComponent(ProfileChooserActivity::class.java.name))
+    }
+  }
+
+  @RunOn(TestPlatform.ESPRESSO) // Testing lifecycle fails on Robolectric.
```

Comment:
Alternatively, you could verify that the activity is finishing using Activity.isFinishing() which I think is Robolectric-friendly.

Reply:
Thanks,`Activity.isFinishing()` does run on robolectric.